src/Bicep.LangServer/Snippets/SnippetCache.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Text.Json;
+using Bicep.Core.Resources;
+
+namespace Bicep.LanguageServer.Snippets;
+
+public record SnippetCache(
+    ImmutableDictionary<ResourceTypeReference, SnippetCache.SnippetCacheSnippet> ResourceTypeReferenceInfoMap,
+    ImmutableDictionary<ResourceTypeReference, string> ResourceTypeReferenceToDependentsMap,
+    ImmutableDictionary<ResourceTypeReference, ImmutableArray<ResourceTypeReference>> ResourceTypeReferenceToChildTypeSymbolsMap,
+    ImmutableArray<Snippet> TopLevelNamedDeclarationSnippets)
+{
+    private static JsonSerializerOptions GetSerializerOptions()
+    {
+        var options = new JsonSerializerOptions();
```

Comment:
Should we use `JavaScriptEncoder.UnsafeRelaxedJsonEscaping`? The "'" characters in the cache file are serialized as `\u0027`.

Reply:
Good point